addon/-private/system/snapshot.js

```diff
@@ -65,7 +65,12 @@ export default class Snapshot {
      */
     this.modelName = internalModel.modelName;
 
-    this._changedAttributes = internalModel.changedAttributes();
+    if (internalModel.isLoading()) {
```

Comment:
It seems like this logic belongs in `internalModel`.

Reply:
@igorT seems fine; you'd prefer to push this into `internalModel.changedAttributes()`?

Reply:
I don't think it is though --- if you have an internal model for a `findRecord` (ie where `isLoading() && !isReloading`), what does it even mean conceptually to ask for changed attributes?

Reply:
If I reorder that you may see what I mean a little bit better: "logic specific to requests"